neuro_extras/image.py

```diff
@@ -380,6 +383,36 @@ async def _build_image(
             raise click.ClickException(f"Failed to build image: {exit_code}")
 
 
+async def _select_build_preset(client) -> Optional[str]:
+    """
+    Try to automatically select the best available preset for image build.
+    """
+    # Require minimum 2 vCPU
+    min_cpu = 2
+    # Require minimum 4 GB RAM
+    min_mem = 4096
+    good_presets = []
+    # Build a shortlist of presets that
+    for cluster_preset_name, cluster_preset_info in client.presets.items():
+        # Don't even try to use GPU machines for image builds
+        if (
+            cluster_preset_info.cpu >= min_cpu
+            and cluster_preset_info.memory_mb >= min_mem
+            and cluster_preset_info.gpu is None
+        ):
+            good_presets.append((cluster_preset_name, cluster_preset_info))
+    if len(good_presets) > 0:
+        # Sort presets by price, then by mem, then by cpu
+        good_presets.sort(
+            key=lambda p: (p[1].credits_per_hour, p[1].memory_mb, p[1].cpu)
+        )
+        preset_name = good_presets[0][0]
+        logger.info(f"Automatically selected build preset {preset_name}")
+        return preset_name
+    else:
+        return None
```

Comment:
Consider the following logic:
- if the user specifies the preset - we validate
  - if it does not satisfy requirements - we print a warning like "The selected resource preset does not satisfy recommended minimum hardware requirements. Consider using 'the-cheapest-one'"
  - if all is good - silently proceed
- if the preset was not specified - we select
  -  if nothing good was found, a warning like "No resource preset satisfying minimal hardware requirements was found in the cluster. The job might take a long time to accomplish. Consider contacting your cluster manager or admin to adjust the cluster configuration.".
  - if one was found - just info "Using 'the-cheapest-one' to run the workload.

How do you feel about this?